src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -141,7 +141,20 @@ void Terminal::UpdateAppearance(const ICoreAppearance& appearance)
     renderSettings.SetRenderMode(RenderSettings::Mode::IntenseIsBold, appearance.IntenseIsBold());
     renderSettings.SetRenderMode(RenderSettings::Mode::IntenseIsBright, appearance.IntenseIsBright());
 
-    switch (appearance.AdjustIndistinguishableColors())
+    // If we're in high contrast mode and AIC is set to Automatic Indexed/Always,
+    // fallback to Indexed/Always respectively.
+    const auto deducedAIC = [this, &appearance]() {
+        const auto initialAIC = appearance.AdjustIndistinguishableColors();
+        switch (initialAIC)
+        {
+        case AdjustTextMode::Automatic:
+            return _highContrastMode ? AdjustTextMode::Indexed : AdjustTextMode::Never;
+        default:
+            return initialAIC;
+        }
+    }();

```

Comment:
Why use a lambda as a scope? You can just use regular `{}` if you want to scope it.